### PR TITLE
HPCC-30680 Documentation covering platform security manager configuration

### PR DESCRIPTION
**Pull request by kenrowland**
…tion

Added document covering the LDAP security manager

Signed-Off-By: Kenneth Rowland kenneth.rowland@lexisnexisrisk.com

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: devdoc/SecurityConfig.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+target. Differences based on deployment can be found in the relevant platform documents.
+
+## Security Managers ##
+Security is implemented via a security manager interface. Managers are loaded and used by components within the
+system to check authorization and authentication. LDAP is an exception to the loadable manager model. It is not a compliant
+loadable module like other security plugins. For that reason, the configuration for each is separated into two 
+sections below: LDAP and Plugin Security Managers.
+
+### LDAP ###
+LDAP is a protocol that connects to an Active Driectory server (AD). The term LDAP is used
```
Comment (JamesDeFabia) [inline devdoc/SecurityConfig.md:21]: Spelling: Directory

File: devdoc/SecurityConfig.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+interchangeably with AD. Below are the configuration values for an LDAP connection. These are valid for both legacy
+(environment.xml) and containerized deployments. For legacy deployments the configuration manager is the 
+primary vehicle for setting these values. However, some values are not available through the tool and must be set 
+manually in the environment.xml if needed for a legacy deployment.
+
+In containerized environments, a ldap configuration block is required for each component. Currently, this results in
```
Comment (JamesDeFabia) [inline devdoc/SecurityConfig.md:27]: LDAP s/b all caps

**Review by JamesDeFabia — COMMENTED** [review]
A few comments inline

File: devdoc/SecurityConfig.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# Security Configuration #
+This document covers security configuration values and meanings. It does not server as
+the source for how to configure security, but rahter what the different values mean. These
```
Comment (afishbeck) [inline devdoc/SecurityConfig.md:3]: "rather"

File: devdoc/SecurityConfig.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+## Security Managers ##
+Security is implemented via a security manager interface. Managers are loaded and used by components within the
+system to check authorization and authentication. LDAP is an exception to the loadable manager model. It is not a compliant
+loadable module like other security plugins. For that reason, the configuration for each is separated into two 
+sections below: LDAP and Plugin Security Managers.
+
+### LDAP ###
+LDAP is a protocol that connects to an Active Driectory server (AD). The term LDAP is used
+interchangeably with AD. Below are the configuration values for an LDAP connection. These are valid for both legacy
```
Comment (afishbeck) [inline devdoc/SecurityConfig.md:22]: Just a side comment, but "interchangeably with AD." I guess is true but that's a bit unfortunate.  I wish we could use other LDAP backends especially if we could have a stand alone one that easily worked on a single developer laptop helm deployment.

File: devdoc/SecurityConfig.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+search for _resourcesBaseDn_
+2. Allowed values for _serverType_ are **ActiveDirectory**, **AzureActiveDirectory**, **389DirectoryServer**, 
+**OpenLDAP**, **Fedora389**
+3. For AzureAD, users are managed from the AD dashboard, not via ECLWatch or through LDAP
+4. If present, _ldapAdminVaultId_ is read and _systemCommonName_ and _systemPassword_ are read from the 
+Kubernetes secrets store and not from the LDAP config values
+5. Must be configured manually in the environment.xml in legacy environments
+
+### Plugin Security Managers ###
+Plugin security managers are separate shared object loaded and intialized by the system. The manager interface is
```
Comment (afishbeck) [inline devdoc/SecurityConfig.md:71]: objects

**Comment by afishbeck** [comment]
This is great for filling in some documentation holes.

**Review by JamesDeFabia — APPROVED** [review]
Looks OK from my POV

**Comment by ghalliday** [comment]
@kenrowland please squash and I will merge

**Comment by kenrowland** [comment]
@ghalliday Please merge

**Pull request merged**